src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudMultipleMatchingConfigurationsIT.java

```diff
@@ -125,4 +128,22 @@ private boolean checkOneNode(PlannedNode plannedNode, String desc) {
         Node node = jenkinsRule.jenkins.getNode(name);
         return desc.equals(node.getNodeDescription());
     }
+
+    private void assertDescription(PlannedNode plannedNode, String agentDesc) throws IOException {
+        String agentName = plannedNode.displayName;
+
+        assertEquals(
+                "Jenkins agent description is incorrect",
+                agentDesc,
+                jenkinsRule.jenkins.getNode(agentName).getNodeDescription());
+
+        // verify the corresponding instance is actually provisioned in GCP
+        // and also has matching description
+        await("Instance is in running status").timeout(2, TimeUnit.MINUTES).until(() -> "RUNNING"
+                .equals(client.getInstance(PROJECT_ID, ZONE, agentName).getStatus()));
+        assertEquals(
+                "Instance description is incorrect",
+                agentDesc,
+                client.getInstance(PROJECT_ID, ZONE, agentName).getDescription());
+    }
```

Comment:
additional assertions ensure that the description actually put on the VM itself too (not just the jenkins agent)